From 522c9b60d3e0a35d29ce8c343226195397f4446e Mon Sep 17 00:00:00 2001
Message-Id: <522c9b60d3e0a35d29ce8c343226195397f4446e.1430272235.git.chang-joon.lee@intel.com>
In-Reply-To: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
References: <0282bce11918ec0dd0c4e7e3bd54e392c2b27a51.1430272235.git.chang-joon.lee@intel.com>
From: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Date: Tue, 14 Apr 2015 16:36:10 +0530
Subject: [PATCH 03/10] FOR_UPSTREAM [VPG]: drm/i915: flis dsi HSI TX
 enable/disable

flis dsi HSI TX bit is cleared during the device_ready and set
during the clear_device_ready.

Issue: GMINL-8141
Change-Id: I97c56ea9643fe8f0f2cd3eecb8cc1e1018c57f9c
Signed-off-by: Yogesh Mohan Marimuthu <yogesh.mohan.marimuthu@intel.com>
Signed-off-by: Uma Shankar <uma.shankar@intel.com>
Signed-off-by: Deepak M <m.deepak@intel.com>
---
 drivers/gpu/drm/i915/intel_dsi.c |   10 ++++++++++
 drivers/gpu/drm/i915/intel_dsi.h |    2 ++
 2 files changed, 12 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_dsi.c b/drivers/gpu/drm/i915/intel_dsi.c
index 95389a5..2f696be 100644
--- a/drivers/gpu/drm/i915/intel_dsi.c
+++ b/drivers/gpu/drm/i915/intel_dsi.c
@@ -218,6 +218,12 @@ static void intel_dsi_device_ready(struct intel_encoder *encoder)
 
 	/* wait for LP state to go 00 */
 	usleep_range(2500, 3000);
+
+	val = vlv_flisdsi_read(dev_priv, FLIS_DSI_TXCNTRL);
+	vlv_flisdsi_write(dev_priv, FLIS_DSI_TXCNTRL,
+				val & ~(BIT_FILS_DSI_TXCNTRL_HS_IO_SEL));
+
+
 	I915_WRITE(MIPI_PORT_CTRL(0), val | LP_OUTPUT_HOLD);
 	usleep_range(1000, 1500);
 	intel_dsi_write_dev_rdy_on_A_and_C(encoder,
@@ -605,6 +611,10 @@ static void intel_dsi_clear_device_ready(struct intel_encoder *encoder)
 	/* wait for dsi controller to be off */
 	usleep_range(2000, 2500);
 
+	val = vlv_flisdsi_read(dev_priv, FLIS_DSI_TXCNTRL);
+	vlv_flisdsi_write(dev_priv, FLIS_DSI_TXCNTRL,
+				val | BIT_FILS_DSI_TXCNTRL_HS_IO_SEL);
+
 	vlv_disable_dsi_pll(encoder);
 }
 
diff --git a/drivers/gpu/drm/i915/intel_dsi.h b/drivers/gpu/drm/i915/intel_dsi.h
index a943dee..f3b8c99 100644
--- a/drivers/gpu/drm/i915/intel_dsi.h
+++ b/drivers/gpu/drm/i915/intel_dsi.h
@@ -398,6 +398,8 @@
 #define LPIO_PWM_ENABLE_MASK		0x80000000
 
 #define FLIS_RCOMP_IOSFSB_REG4		0x04
+#define FLIS_DSI_TXCNTRL		0x15
+#define BIT_FILS_DSI_TXCNTRL_HS_IO_SEL	(1 << 0)
 
 struct intel_dsi_device {
 	unsigned int panel_id;
-- 
1.7.9.5

